package/gluon-state/luasrc/usr/sbin/gluon-state

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/lua
+
+local bit = require 'bit'
+local util = require 'gluon.util'
+local unistd = require 'posix.unistd'
+local stat = require 'posix.sys.stat'
+
+local state_dir = "/var/gluon/state/"
+local check_dir = "/lib/gluon/state/check.d/"
+
+
+local function set_flag(stateflag, state)
+	if state then
+		-- this does not modify atime
+		local flaghandle = io.open(stateflag, "w")
+		flaghandle:close()
+	else
+		os.remove(stateflag)
+	end
+end
+
+local function exec_check(checkpath)
+	local checkname = string.sub(checkpath, #check_dir+1)
+	local ret = os.execute(checkpath)
+	local flagfile = state_dir..checkname
+	set_flag(flagfile, 0==ret)
+end
+
+local function run_executable_checks()
+	for _, v in ipairs(util.glob(check_dir..'*')) do
+		local statmode = stat.stat(v).st_mode
+		local is_file = statmode ~= nil and stat.S_ISREG(statmode)
+		if is_file and bit.band(statmode, stat.S_IXUSR) ~= 0 then
+			exec_check(v)
+		end
+	end
+end
+
+
+-- ensure state path exists
+if not unistd.access(state_dir) then
+	stat.mkdir(state_dir)
+end
```

Comment:
```suggestion
-- ensure state path exists
if not unistd.access(state_dir) then
	mkdir_recursive(state_dir)
end
```

While you ensure that `/var/gluon/state` is created, you do not ensure that `/var/gluon/` is already created. `/var/gluon` is only created [here](https://github.com/freifunk-gluon/gluon/blob/d3d22ba677f3547c105c05fd5e66e80453e4b824/package/gluon-wan-dnsmasq/luasrc/lib/gluon/wan-dnsmasq/update.lua). Let's not create an implicit dependency here.

As we ignore the return values of *stat.mkdir()* anyways, checking for the existence with *unistd.access()* is unnecessary.

Reply:
outdated since 46b32d8a5a74562de174c907fcc53eac75de1d0e